src/plugins/main.js

```diff
@@ -30,14 +30,18 @@ export default async (context) => {
       namespaced: true,
       state: () => ({
         locale: '',
-        messages: {}
+        messages: {},
+        routeParams: {}
       }),
       actions: {
         setLocale ({ commit }, locale) {
           commit(vuex.mutations.setLocale, locale)
         },
         setMessages ({ commit }, messages) {
           commit(vuex.mutations.setMessages, messages)
+        },
+        setRouteParams ({ commit }, params) {
+          commit(vuex.mutations.setRouteParams, params)
```

Comment:
I guess you have access to all locales here so it could be nice to validate keys of the object here according to available locale. That's because I feel people might commit this object with wrong structure and then will wonder why things are not working. Just a suggestion...